File: cap07/src/main/java/app/car/cap07/interfaces/incoming/errorhandling/LocaleResolver.java
package app.car.cap07.interfaces.incoming.errorhandling;

import org.apache.commons.lang3.StringUtils;
import org.springframework.stereotype.Component;
import org.springframework.web.servlet.i18n.AcceptHeaderLocaleResolver;

import jakarta.servlet.http.HttpServletRequest;
import java.util.Arrays;
import java.util.List;
import java.util.Locale;
import java.util.Optional;

@Component
public class LocaleResolver extends AcceptHeaderLocaleResolver {

    private static final Locale DEFAULT_LOCALE = Locale.of("pt", "BR");

    private static final List<Locale> ACCEPTED_LOCALES = List.of(DEFAULT_LOCALE, Locale.of("en"));

    @Override
    public Locale resolveLocale(HttpServletRequest request) {
        final String acceptLanguageHeader = request.getHeader("Accept-Language");
        if (StringUtils.isBlank(acceptLanguageHeader) || acceptLanguageHeader.trim().equals("*")) {
            return DEFAULT_LOCALE;
        }
        List<Locale.LanguageRange> list = Locale.LanguageRange.parse(acceptLanguageHeader);
        Locale locale = Locale.lookup(list, ACCEPTED_LOCALES);
        return Optional.ofNullable(locale).orElse(DEFAULT_LOCALE);
    }


}

File: cap09/src/main/kotlin/app/car/car09/interfaces/incoming/errorhandling/LocaleResolver.kt
package app.car.car09.interfaces.incoming.errorhandling

import org.springframework.stereotype.Component
import org.springframework.web.servlet.i18n.AcceptHeaderLocaleResolver
import java.util.Locale
import java.util.Locale.LanguageRange
import javax.servlet.http.HttpServletRequest


@Component
class LocaleResolver : AcceptHeaderLocaleResolver() {

    override fun resolveLocale(request: HttpServletRequest): Locale {
        val acceptLanguageHeader = request.getHeader("Accept-Language")
        if (acceptLanguageHeader.isBlank() || acceptLanguageHeader.trim() == "*") {
            return DEFAULT_LOCALE
        }
        val list = LanguageRange.parse(acceptLanguageHeader)
        val locale = Locale.lookup(list, ACCEPTED_LOCALES)
        return locale ?: DEFAULT_LOCALE
    }

    companion object {
        private val DEFAULT_LOCALE = Locale("pt", "BR")
        private val ACCEPTED_LOCALES = listOf(DEFAULT_LOCALE,Locale("en"))
    }
}
